securebanking-openbanking-uk-gateway-conversions/src/main/java/com/forgerock/securebanking/uk/gateway/conversion/converters/AccountAccessIntentConverter.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.securebanking.uk.gateway.conversion.converters;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.org.openbanking.datamodel.account.OBReadConsentResponse1;
+
+public class AccountAccessIntentConverter extends GenericIntentConverter<OBReadConsentResponse1, String> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AccountAccessIntentConverter.class);
+
+    public AccountAccessIntentConverter() {
+        super(AccountAccessIntentConverter::convertToOBObject);
+    }
+
+    private static OBReadConsentResponse1 convertToOBObject(String jsonString) {
```

Comment:
I'd make this an instance method and use the objectMapper field in the base class. Otherwise this is creating a new mapper obj each time it is invoked.

Alternatively, instead of passing a conversion Function to the base class, it could have an abstract convertToObObject method which subclasses (AccountAccessIntentConverter etc) would implement.

Reply:
Actually, probably just need to review the class hierarchy a little. The base class has a mapper field but it never uses it itself.

Reply:
Deleted `mapper` field from generic class, to be keep a static method to get the generic converter mapper to be used statically in tests and in the static methods of class hierarchy.
A hierarchy class can get the mapper using the static method `generiConverterMapper()` and add new configurations if necessary before call the `readValue(...)` mapper method to map the content.

Reply:
Deleted mapper field from generic Converter class and moved to a Generic mapper singleton class.